x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/endpoint_details.tsx

```diff
@@ -4,207 +4,223 @@
+        <EuiLoadingContent lines={3} />
       </>
-    );
-  }
-);
+    ),
+    []
+  );
+
+  const getTabs = useCallback(
+    (id: string) => [
+      {
+        id: EndpointDetailsTabsTypes.overview,
+        name: i18.OVERVIEW,
+        route: getEndpointDetailsPath({
+          ...queryParams,
+          name: 'endpointDetails',
+          selected_endpoint: id,
+        }),
+        content:
+          hostDetails === undefined ? (
+            ContentLoadingMarkup
+          ) : (
+            <EndpointDetailsContent
+              details={hostDetails}
+              policyInfo={policyInfo}
+              hostStatus={hostStatus}
+            />
+          ),
+      },
+      {
+        id: EndpointDetailsTabsTypes.activityLog,
+        name: i18.ACTIVITY_LOG.tabTitle,
+        route: getEndpointDetailsPath({
+          ...queryParams,
+          name: 'endpointActivityLog',
+          selected_endpoint: id,
+        }),
+        content: <EndpointActivityLog activityLog={activityLog} />,
+      },
+    ],
+    [ContentLoadingMarkup, hostDetails, policyInfo, hostStatus, activityLog, queryParams]
+  );
+
+  const showFlyoutFooter =
+    show === 'details' || show === 'policy_response' || show === 'activity_log';
+
+  useEffect(() => {
+    if (hostDetailsError !== undefined) {
+      toasts.addDanger({
+        title: i18n.translate('xpack.securitySolution.endpoint.details.errorTitle', {
+          defaultMessage: 'Could not find host',
+        }),
+        text: i18n.translate('xpack.securitySolution.endpoint.details.errorBody', {
+          defaultMessage: 'Please exit the flyout and select an available host.',
+        }),
+      });
+    }
+  }, [hostDetailsError, show, toasts]);
+  return (
+    <>
+      {(show === 'policy_response' || show === 'isolate' || show === 'unisolate') && (
+        <EndpointDetailsFlyoutHeader hostname={hostDetails?.host?.hostname} />
+      )}
+      {hostDetails === undefined ? (
+        <EuiFlyoutBody>
+          <EuiLoadingContent lines={3} /> <EuiSpacer size="l" /> <EuiLoadingContent lines={3} />
+        </EuiFlyoutBody>
+      ) : (
+        <>
+          {(show === 'details' || show === 'activity_log') && (
+            <EndpointDetailsFlyoutTabs
+              hostname={hostDetails.host.hostname}
+              show={show}
+              tabs={getTabs(hostDetails.agent.id)}
+            />
+          )}
+
+          {show === 'policy_response' && <PolicyResponseFlyoutPanel hostMeta={hostDetails} />}
+
+          {(show === 'isolate' || show === 'unisolate') && (
+            <EndpointIsolationFlyoutPanel hostMeta={hostDetails} />
+          )}
+
+          {showFlyoutFooter && (
+            <EuiFlyoutFooter className="eui-textRight" data-test-subj="endpointDetailsFlyoutFooter">
+              <ActionsMenu />
+            </EuiFlyoutFooter>
+          )}
+        </>
+      )}
+    </>
+  );
+});
 
 EndpointDetails.displayName = 'EndpointDetails';
+
+const PolicyResponseFlyoutPanel = memo<{
```

Comment:
Would be good to move this component outside this file, it will keep this file small and a better test file organisation

Reply:
Yes. I will do that in the following PR when I move it to a common component. For now this is simply taking things out of the flyout.